### PR TITLE
Convert `<top-bar>` to a component

### PR DESCRIPTION
**Pull request by robertknight**
_This is part of the effort to convert all of the UI elements to components. See https://github.com/hypothesis/client/pull/292_

Convert `<top-bar>` to a component. This involves:

- Moving properties from the scope to the controller, so they are accessed via `vm.thing` instead of `thing` in the template.
- Simplifying the exports of `top-list.js` and moving it to `src/components`

### CODE REVIEW

File: src/sidebar/templates/top_bar.html [hunk 1]
```diff
@@ -23,39 +23,39 @@
...
     <login-control
       class="login-control"
-      auth="auth"
+      auth="vm.auth"
       new-style="true"
-      on-show-help-panel="onShowHelpPanel()"
-      on-login="onLogin()"
+      on-show-help-panel="vm.onShowHelpPanel()"
+      on-login="vm.onLogin()"
       on-logout="onLogout()">
```
Comment (sean-roberts) [inline src/sidebar/templates/top_bar.html]: missed vm.onLogout?

Comment (robertknight) [inline src/sidebar/templates/top_bar.html]: 🤕 - Gah, sloppy of me. I'll get this fixed and look into adding some tests or other form of checks.

**Comment by robertknight** [comment]
Rebased

**Comment by robertknight** [comment]
Rebased

**Comment by sean-roberts** [comment]
Good to go, I will leave it up to you to merge since you are in the middle of deciding a release

**Pull request merged**